### ISSUE TITLE
Compilation fails because of unused variable

### ISSUE THREAD

**Issue by salokanto** [alisaifee/asdf-thrift#3]
Hi, the following happens on both M1 and Intel Macs:

```
❯ asdf install thrift 0.13.0
Downloading source.....
Configuring build....
Compiling with flags -j16....
/Users/foo/.asdf/installs/thrift/0.13.0/src/compiler/cpp/src/thrift/thrifty.yy:1.1-5: warning: POSIX Yacc does not support %code [-Wyacc]
    1 | %code requires {
      | ^~~~~
src/thrift/generate/t_java_generator.cc:5369:9: error: variable 'j' set but not used [-Werror,-Wunused-but-set-variable]
    int j = 0;
        ^
1 error generated.
make[2]: *** [src/thrift/generate/thrift-t_java_generator.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [install-recursive] Error 1
make: *** [install-recursive] Error 1
asdf-thrift: An error ocurred while installing thrift 0.13.0. See /var/folders/lf/dtnk7rp90rgc24_9pr6n8rkh0000gn/T//thrift-install-0.13.0.txt
```

Getting the same error for at least Thrift versions 0.13.0, 0.15.0, and 0.18.1 -- except that the line number for the unused variable is different.

Something like `-Wno-unused-variable` could bypass this, but I couldn't find how to pass options to asdf build.

**System:**

MacOS Ventura 13.4.

```
❯ gcc --version
Apple clang version 14.0.3 (clang-1403.0.22.14.1)
Target: x86_64-apple-darwin22.5.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

**Comment by jmberrueta** [comment]
Same here!

Also, on an M1, it seems default pkg-config was not good enough. Had to install a new one with `brew install pkg-config` (might be worth documenting)

**Comment by jmberrueta** [comment]
I looked a bit more into this.

It seems the error due to the warning can be disabled by adding the `-Wno-unused-but-set-variable` CXXFLAG at  https://github.com/alisaifee/asdf-thrift/blob/master/lib/utils.bash#L49:
```
export CXXFLAGS="-Wno-inconsistent-missing-override -Wno-unused-but-set-variable"
```
(in MacOS the plugin file can be found at `~/.asdf/plugins/thrift/lib/utils.bash` in case you wanna change that locally)

However after doing that I'm facing yet a new error:
```
src/thrift/generate/t_cl_generator.cc:29:10: fatal error: 'boost/tokenizer.hpp' file not found
#include <boost/tokenizer.hpp>
```
I already had installed `boost` with brew, but that does not seem to be enough. Adding the `-I/opt/homebrew/include` CXXFLAG did the trick here. Again, I did this by changing my local `utils.bash`, and this is assuming you installed boost through via Homebrew:
```
export CXXFLAGS="-Wno-inconsistent-missing-override -Wno-unused-but-set-variable -I/opt/homebrew/include"
```

**Comment by johnjelinek** [comment]
Here, have a patch:
```diff
diff --git a/lib/utils.bash b/lib/utils.bash
index a2aff74..d649bbe 100644
--- a/lib/utils.bash
+++ b/lib/utils.bash
@@ -47,7 +47,7 @@ install_version() {
 		local bison_brew_path
 		bison_brew_path="$(brew --prefix bison)/bin"
 		export PATH="$bison_brew_path:$PATH"
-		export CXXFLAGS="-Wno-inconsistent-missing-override"
+		export CXXFLAGS="${CXXFLAGS:--Wno-inconsistent-missing-override}"
 		local bison_version
 		if ! type "automake" >/dev/null 2>&1; then
 			fail "Thrift requires automake"

```
Tested like this:
* keeps existing behavior
```shell
$ asdf install thrift 0.18.0

Downloading source.....
Configuring build....
Compiling with flags -j10....
${HOME}/.asdf/installs/thrift/0.18.0/src/compiler/cpp/src/thrift/thrifty.yy:1.1-5: warning: POSIX Yacc does not support %code [-Wyacc]
    1 | %code requires {
      | ^~~~~
src/thrift/generate/t_java_generator.cc:5754:9: error: variable 'j' set but not used [-Werror,-Wunused-but-set-variable]
    int j = 0;
        ^
1 error generated.
make[2]: *** [src/thrift/generate/thrift-t_java_generator.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [install-recursive] Error 1
make: *** [install-recursive] Error 1
asdf-thrift: An error ocurred while installing thrift 0.18.0.
```
* works on M1
```shell
$ env CXXFLAGS="-Wno-inconsistent-missing-override -Wno-unused-but-set-variable -I$(brew --prefix)/include" asdf install thrift 0.18.0

Downloading source.....
Configuring build....
Compiling with flags -j10....
${HOME}/.asdf/installs/thrift/0.18.0/src/compiler/cpp/src/thrift/thrifty.yy:1.1-5: warning: POSIX Yacc does not support %code [-Wyacc]
    1 | %code requires {
      | ^~~~~
ld: warning: -bind_at_load is deprecated on macOS
```

**Comment by johnjelinek** [comment]
@alisaifee: can we get a release with this? 👆

**Comment by alisaifee** [comment]
Resolved in fecdd6c3b081ef499527ed0aa03fbab2374820dd

**Comment by johnjelinek** [comment]
You may want to add some documentation in the readme about passing in the CXXFLAGS env var to build on M1
